### PR TITLE
Fix S3 GetObject header boolean value in uppercase (#6840)

### PR DESCRIPTION
**Pull request by ChandanChainani**
* Some AWS SDKs require header flag value in bool to be in lowercase

**Please refer to the contribution guidelines in the README when submitting PRs.**

### CODE REVIEW

**Comment by ChandanChainani** [comment]
recheck

**Comment by ChandanChainani** [comment]
recheck

**Comment by ChandanChainani** [comment]
recheck

**Comment by ChandanChainani** [comment]
recheck

**Comment by ChandanChainani** [comment]
@localstack-bot recheck

**Comment by ChandanChainani** [comment]
recheck

**Comment by ChandanChainani** [comment]
@localstack-bot recheck

**Comment by ChandanChainani** [comment]
> 

recheck

**Comment by ChandanChainani** [comment]
recheck

**Comment by ChandanChainani** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by thrau — COMMENTED** [review]
Thanks for the contribution @ChandanChainani!

I'm a bit fussy about patching framework fundamentals, and especially about touching header casing. Manipulating header casing keeps causing us incredible headaches, so I would prefer not to make this broad of a patch. Moreover, technically the `Headers` does not allow boolean values in the first place.

It seems better to fix the root of the problem, which seems to be here in moto: https://github.com/spulec/moto/blob/3118090fdc38de64e0a7aef969b3b1992c1703f5/moto/s3/models.py#L248-L250 I think we should just change the line to `str(self.bucket_key_enabled).lower()` there.

Moving forward this shouldn't be a problem at all, since we are currently migrating s3 to the new framework, which will serialize the response types properly into an HTTP response, which covers boolean to string conversions. But for now, it seems moto is the best place to make this change.

**Comment by ChandanChainani** [comment]
@thrau I was thinking the same but then I though there may be other headers like `x-amz-server-side-encryption-bucket-key-enabled` in `s3` or other services which possibly may have the same behaviour that why instead of changing in moto I create a wrapper that will change in all responses.

File: tests/integration/s3/test_s3.py [hunk 1]
```diff
@@ -2167,6 +2167,27 @@ def test_s3_get_object_header_overrides(self, s3_client, s3_bucket, snapshot):
...
+                            "KMSMasterKeyID": "secret",
+                        }
+                    },
+                ]
+            },
+        )
+        response = s3_client.put_object(Bucket=bucket_name, Key="test-key", Body="test-value")
+        assert response["ResponseMetadata"]["HTTPHeaders"][
+            "x-amz-server-side-encryption-bucket-key-enabled"
+        ] in ["false", "true"]
```
Comment (thrau) [inline tests/integration/s3/test_s3.py:2189]: I think we should be stronger with the assertion here. Which value should it be in this case? true or false?

**Review by thrau — COMMENTED** [review]
Thanks @ChandanChainani for explaining your reasoning. My argument still stands, in that I think adding non-string values to the Headers object should be treated as a bug. 

I'm happy to keep the regression test though, and would be very grateful for a fix of the bug upstream in moto!

**Comment by bentsku** [comment]
Hello @ChandanChainani and sorry for the delay again, thanks again for the contribution.

Do you still encounter the issue? Since then, we moved to the new implementation of S3 which shouldn't have this bug anymore. If it's working for you, I think we could close this PR.

**Comment by bentsku** [comment]
I'll close the PR now. If you still encounter the issue, feel free to open an issue describing it. Thanks again for your contribution!

**Pull request closed without merging**